src/Type/Php/ReplaceFunctionsDynamicReturnTypeExtension.php

```diff
@@ -9,21 +9,23 @@
 use PHPStan\Type\ArrayType;
 use PHPStan\Type\DynamicFunctionReturnTypeExtension;
 use PHPStan\Type\MixedType;
+use PHPStan\Type\NullType;
 use PHPStan\Type\StringType;
 use PHPStan\Type\Type;
 use PHPStan\Type\TypeUtils;
+use PHPStan\Type\UnionType;
 
 class ReplaceFunctionsDynamicReturnTypeExtension implements DynamicFunctionReturnTypeExtension
 {
 
 	/** @var array<string, int> */
 	private $functions = [
-		'preg_replace' => 2,
-		'preg_replace_callback' => 2,
-		'preg_replace_callback_array' => 1,
-		'str_replace' => 2,
-		'str_ireplace' => 2,
-		'substr_replace' => 0,
+		'preg_replace' => ['argumentPosition' => 2, 'returnNull' => true],
```

Comment:
I don't think you need this. You can use `TypeCombinator::containsNull($functionReflection->getReturnType())`.

Reply:
If those functions do not have `null` in `functionMap.php`, you should add it there.

Reply:
Fixed